src/routes/snippets.rs

```diff
@@ -21,6 +24,57 @@ fn create_snippet_impl(
     Ok(Created(location, Some(Output(new_snippet))))
 }
 
+fn create_link_header(
+    origin: &Origin,
+    next_marker: Option<String>,
+    prev_marker: Option<String>,
+    prev_needed: bool,
+) -> String {
+    let query_wo_marker = origin.query().map(|q| {
+        q.split("&")
+            .map(|v| v.split("=").collect::<Vec<&str>>())
+            .filter(|v| v[0] != "marker")
+            .map(|v| v.join("="))
+            .collect::<Vec<String>>()
+            .join("&")
+    });
+    let query_first = query_wo_marker.clone();
+    let mut query_next = next_marker.map(|marker| format!("marker={}", marker));
+    let mut query_prev = prev_marker.map(|marker| format!("marker={}", marker));
+
+    // If a request URL does contain query parameters (other than marker), we
+    // must reuse them together with next/prev markers.
+    if query_wo_marker.is_some() {
+        query_next =
+            query_next.map(|query| format!("{}&{}", query_wo_marker.clone().unwrap(), query));
```

Comment:
> you shouldn't need clone() here either, as format!() already allocates a new String

I don need `.clone()` because `.unwrap()` consumes the value and I don't want it to be consumed. With `if let` approach, however, it might not be needed anymore though.

Reply:
Ok :) It's a bit verbose because we need to push '&' and the value of the marker separately:

```diff
     // If a request URL does contain query parameters (other than marker), we
     // must reuse them together with next/prev markers.
-    if query_wo_marker.is_some() {
+    if let Some(query_wo_marker) = query_wo_marker {
         query_next =
-            query_next.map(|query| format!("{}&{}", query_wo_marker.clone().unwrap(), query));
+            query_next.map(|mut query| { query.push('&'); query.push_str(&query_wo_marker); query });
         query_prev =
-            query_prev.map(|query| format!("{}&{}", query_wo_marker.clone().unwrap(), query));
+            query_prev.map(|mut query| { query.push('&'); query.push_str(&query_wo_marker); query });
     }
```